File: 25-03-2024_Print_N-bit_binary_numbers_having_more_1s_than_0s.cpp
class Solution{
public:	

    void generate(int n, string s, int ones, int zeroes, vector<string>& ans){
        if(s.length() == n){
            ans.push_back(s);
            return;
        }
        generate(n, s + "1", ones + 1, zeroes, ans);
        
        if(ones > zeroes){
            generate(n, s + "0", ones, zeroes + 1, ans);
        }
    }

	vector<string> NBitBinary(int n)
	{
	    // Your code goes here
	    vector<string> ans;
	    generate(n, "", 0, 0, ans);
	    
	    return ans;
	}
};

File: 25-03-2024_Print_N-bit_binary_numbers_having_more_1s_than_0s.java
class Solution {
    
    void generate(String str, int ones, int zeroes, ArrayList<String>ans, int n){
        if(str.length() == n){
            ans.add(str);
            return;
        }
        generate(str + "1", ones + 1, zeroes, ans, n);
        if(ones > zeroes){
            generate(str + "0", ones, zeroes + 1, ans, n);
        }
    }
    
    ArrayList<String> NBitBinary(int N) {
        // code here
        ArrayList<String>ans = new ArrayList<>();
        String s = "";
        generate(s, 0, 0, ans, N);
        return ans;
    }
}
